onnx/cpp2py_export.cc

```diff
@@ -26,6 +28,57 @@
 #pragma error "Pybind11 must be >= 2.12 to be compatible with numpy 2.0."
 #endif
 
+template <typename T>
+struct PythonProtoTypeMap {};
+
+#define DEFINE_PROTO_TYPE_MAP(_ProtoType, PY_MODULE_NAME, PY_TYPE_NAME)                             \
```

Comment:
For this part, is it possible to do serialization on the python side? Similar to https://github.com/onnx/onnx/blob/da8d7c1b404ce673c21a41a39ec17df91c80ef17/onnx/shape_inference.py#L48

Would that be cleaner?